src/gwt/src/org/rstudio/studio/client/workbench/views/jobs/JobsPresenter.java

```diff
@@ -185,6 +199,15 @@ public void onActivateJobs()
       display_.bringToFront();
    }
    
+   @Handler
+   public void onHideCompletedJobs()
+   {
+      boolean newValue = !uiPrefs_.hideCompletedJobs().getValue();
+      uiPrefs_.hideCompletedJobs().setGlobalValue(newValue);
+      commands_.hideCompletedJobs().setChecked(newValue);
```

Comment:
Do we need this line? (looks like the command will get its checked state automatically updated from the UI pref update handler)

Reply:
That's correct, I'll remove that. We need a nice way to have main-menu commands that automatically track a uiPref, like we do for manually created menus. Maybe it's there, but I didn't see it.

Reply:
It sounds like you've already seen https://github.com/rstudio/rstudio/blob/master/src/gwt/src/org/rstudio/core/client/widget/UIPrefMenuItem.java for regular menu items? More generally it'd be nice to make the relationship between the command and the pref declarative (so that any rendering and treatment would be correct) -- I think this could live in Commands.cmd.xml and we could generate the right code in the emitter. Not a 1.2 project.